spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -76,6 +76,12 @@ def get_parser():
         help='Concatenate data along the specified dimension',
         required=False,
         choices=('x', 'y', 'z', 't'))
+    image.add_argument(
+        '-stitch',
+        action='store_true',
+        help='Stitch multiple images acquired in the same orientation utilizing '
+             'the algorithm by Lavdas, Glocker et al. (https://doi.org/10.1016/j.crad.2019.01.012).',
+        required=False)
```

Comment:
Because [`build_biomedical_stitching`](https://github.com/jqmcginnis/build_biomedia_stitching) is still a work-in-progress, I would suggest a temporary workaround:

* Build `stitching` locally on 1 OS (I assume you've already done this on your local machine? :wink:)
* Download SCT's other C++ binaries from [this release](https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries/releases), corresponding to your OS.
* Extract the binaries, then add `stitching`, then zip up the folder again.
* Upload it somewhere temporarily
* Change [the link in `sct_download_data`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/bd3d89902106be3186fe051096c72814a95982ce/spinalcordtoolbox/scripts/sct_download_data.py#L77-L82) so that it points to the temporary zip containing `stitching`

If you do that, then `stitching` will be present to be used in SCT's test suite. That way, we can at least create a new test for at least one OS (while we figure out how to build `stitching` for the other OSs).

Reply:
As for actually writing the test, we would go to [`testing/cli/test_cli_sct_image.py`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/testing/cli/test_cli_sct_image.py) and create a new test that runs `sct_image -stitch`.

To limit this test to only 1 OS, we can use the `@pytest.mark.skipif` decorator. For example:

```python
@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="temporarily testing 'stitching' on just linux")
def test_sct_image_stitch():
    # test goes here
```

Reply:
I swapped the link for a binary release including stitched in [7db244e](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/7db244e8876083a0630649aac6e575d936a38a2b) :slightly_smiling_face: 

For implementing the testing function I am assuming we would also need some files to stitch. Do you happen to know if there's any data (e.g. in [spine generic](https://github.com/spine-generic/spine-generic) or [sct_pipeline](https://github.com/sct-pipeline) ) we could use for the test cases, otherwise we could also use a whole spine image and create multiple stacks out of it, and e.g. add some noise to it, to make it a more challenging stitching task?

Thanks again @joshuacwnewton for letting me work on this and helping with all related questions, I am learning a lot :student:

Reply:
> For implementing the testing function I am assuming we would also need some files to stitch. Do you happen to know if there's any data (e.g. in [spine generic](https://github.com/spine-generic/spine-generic) or [sct_pipeline](https://github.com/sct-pipeline) ) we could use for the test cases, otherwise we could also use a whole spine image and create multiple stacks out of it, and e.g. add some noise to it, to make it a more challenging stitching task?

Currently, we use a dataset called `sct_testing_data` for our tests. You can take a look at it by running the following command:

```
sct_download_data -d sct_testing_data
```

This dataset gets downloaded to the `/data/` folder in the SCT installation directory.

However, the files in `sct_testing_data` are not full spinal cord images as you mention (most are cropped cord regions), so I'm not 100% sure if they will be useful for testing `stitching`. 🤔

Even if the images aren't perfect, though, we can still use them temporarily to test the script to make sure everything surrounding `stitching` (argument parsing, file loading and saving, etc.) works as expected. 

> Thanks again @joshuacwnewton for letting me work on this and helping with all related questions, I am learning a lot :student: 

Thank you again for such kind words!! I'm glad to help. 🙂

Reply:
As a test case we could use`sct_crop_image` to crop `t2.nii.gz` into two ROIs and employ `stitching` to merge them again. Do you think this would be an appropriate test? I feel it is not a challenging task, but we probably do not want to check how well the stitching algorithm works, more if it works reliably in the CI/CD pipeline?

E.g. to test this, you can try to run:

```
sct_crop_image -i t2.nii.gz -xmin 0 -xmax 59 -ymin 0 -ymax 20 -zmin 0 -zmax 51 -o sct_roi1.nii.gz
sct_crop_image -i t2.nii.gz -xmin 0 -xmax 59 -ymin 20 -ymax 40 -zmin 0 -zmax 51 -o sct_roi2.nii.gz
sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch
```

Reply:
> I feel it is not a challenging task, but we probably do not want to check how well the stitching algorithm works, more if it works reliably in the CI/CD pipeline?

That is exactly what I was thinking! :smile: 

My understanding was that a proper evaluation of the `stitching` algorithm would be something like... a one-off test on a larger dataset? With the results presented and discussed in, say, the Sept 7th meeting, or something similar to that? 

Then, once we feel confident in the `stitching` algorithm, the CI test would be just for basic functionality, as you say! Mostly because the `stitching` binary is basically a black box, and one that likely won't be updated as frequently as our Python source code.

> ```
> sct_crop_image -i t2.nii.gz -xmin 0 -xmax 59 -ymin 0 -ymax 20 -zmin 0 -zmax 51 -o sct_roi1.nii.gz
> sct_crop_image -i t2.nii.gz -xmin 0 -xmax 59 -ymin 20 -ymax 40 -zmin 0 -zmax 51 -o sct_roi2.nii.gz
> sct_image -i sct_roi1.nii.gz sct_roi2.nii.gz -o stitched.nii.gz -stitch
> ```

Wonderful! This is very easy to convert into a `pytest` test inside `test_cli_sct_image`:

You can use a [`pytest.fixture`](https://docs.pytest.org/en/6.2.x/fixture.html) to contain the setup cropping code, similar to something like this:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2769d530209dc6e62afd32266417bfe4dc1a5897/testing/cli/test_cli_sct_image.py#L20-L38

Reply:
Ah wait, sorry! You already wrote a test. :smile: 

My bad, let me check that over now. :slightly_smiling_face: